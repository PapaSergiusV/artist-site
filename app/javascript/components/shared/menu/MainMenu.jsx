import React from "react";
import PropTypes from "prop-types";
import CSSModules from "react-css-modules";
import { FaInstagram, FaPhone, FaEnvelope, FaTimes } from "react-icons/fa";
import Routes from "../../../libs/routes";

import usePopupAnimation from "../usePopupAnimation";
import styles from "./MainMenu.module.scss";

const MainMenu = props => {
  const { albums = [{ name: "", id: 0 }], contacts: { instagram, phone, email } } = props;
  const [wrapper, close] = usePopupAnimation(props.close);
  return (
    <div styleName="wrapper" ref={wrapper} style={{ opacity: 0 }}>
      <div className="container">
        <div className="row">
          <div className="col-12" styleName="row">
            <div styleName="gold">LOGO</div>
            <div styleName="close-button">
              <FaTimes onClick={close} />
            </div>
          </div>
          <div className="col-12" styleName="row">
            <div>
              <ul>
                <li><a href={Routes.root_path()}>Main</a></li>
                <li><a href={Routes.albums_path()}>Albums</a></li>
                <li><a href="#">About</a></li>
              </ul>
            </div>
            <div>
              {
                albums.map(album =>
                  <p key={album.id}>
                    <a href={Routes.album_path(album.id)}>
                      {album.name}
                    </a>
                  </p>
                )
              }
            </div>
          </div>
          <div className="col-12" styleName="row">
            <hr />
          </div>
          <div className="col-12" styleName="row">
            <div styleName="address-point">
              <FaInstagram />
              <a href={instagram.link}>
                {instagram.title}
              </a>
            </div>
            <div styleName="address-point">
              <FaPhone />
              <span>{phone.link}</span>
            </div>
            <div styleName="address-point">
              <FaEnvelope />
              <span>{email.link}</span>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

const Contact = PropTypes.shape({
  link: PropTypes.string.isRequired,
  title: PropTypes.string
});

MainMenu.propTypes = {
  albums: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired
  })),
  close: PropTypes.func.isRequired,
  contacts: PropTypes.shape({
    instagram: Contact,
    phone: Contact,
    email: Contact
  }).isRequired
};

export default CSSModules(MainMenu, styles, { allowMultiple: true });
