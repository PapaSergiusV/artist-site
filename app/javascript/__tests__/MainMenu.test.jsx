import React from "react";
import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import "../setupTests";

import MainMenu from "../components/shared/menu/MainMenu";

describe("MainMenu component", () => {
  it("should render logo and menu button", () => {
    const wrapper = shallow(<MainMenu albums={[{ name: "1" }, { name: "2" }, { name: "3" }]} />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find(".albums").children().length).toEqual(3);
    const withoutProps = shallow(<MainMenu />);
    expect(withoutProps.find(".albums").children().length).toEqual(1);
  });
});