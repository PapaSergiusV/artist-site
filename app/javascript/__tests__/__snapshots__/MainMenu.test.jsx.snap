// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainMenu component should render logo and menu button 1`] = `
<div
  className="wrapper"
  style={
    Object {
      "opacity": 0,
    }
  }
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-12 row"
      >
        <div
          className="gold"
        >
          LOGO
        </div>
        <div
          className="close-button"
        >
          <FaTimes
            onClick={[Function]}
          />
        </div>
      </div>
      <div
        className="col-12 row"
      >
        <div>
          <ul>
            <li>
              Main
            </li>
            <li>
              Albums
            </li>
            <li>
              About
            </li>
          </ul>
        </div>
        <div
          className="albums"
        >
          <p
            key="1"
          >
            1
          </p>
          <p
            key="2"
          >
            2
          </p>
          <p
            key="3"
          >
            3
          </p>
        </div>
      </div>
      <div
        className="col-12 row"
      >
        <hr />
      </div>
      <div
        className="col-12 row"
      >
        <div>
          <p>
            Address
          </p>
        </div>
        <div>
          <FaInstagram />
          <FaVk />
        </div>
      </div>
    </div>
  </div>
</div>
`;
